components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java

```diff
@@ -152,6 +152,46 @@ public OMElement readAndCleanWsdl2(API api) throws APIManagementException {
 
     }
 
+    /**
+     * Gets WSDL definition as a byte array
+     *
+     * @return converted WSDL definition as byte array
+     * @throws APIManagementException
+     */
+    public byte[] getWSDL() throws APIManagementException {
+        try {
+            Definition wsdlDefinition = readWSDLFile();
+            WSDLWriter writer = getWsdlFactoryInstance().newWSDLWriter();
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            writer.writeWSDL(wsdlDefinition, byteArrayOutputStream);
+            return byteArrayOutputStream.toByteArray();
+        } catch (Exception e) {
+            String msg = "Error occurs when change the address URL of the WSDL";
+            throw new APIManagementException(msg, e);
+        }
+    }
+
+    /**
+     * Returns WSDL definition from a byte content of the WSDL
+     *
+     * @param wsdl byte content of the WSDL document
+     * @return {@link Definition} - WSDL4j definition constructed form the wsdl
+     * @throws APIManagementException
+     */
+    public Definition getWSDLDefinitionFromByteContent(byte[] wsdl) throws APIManagementException {
+        try {
+            WSDLReader wsdlReader = getWsdlFactoryInstance().newWSDLReader();
+            // switch off the verbose mode
+            wsdlReader.setFeature(JAVAX_WSDL_VERBOSE_MODE, false);
+            wsdlReader.setFeature("javax.wsdl.importDocuments", false);

```

Comment:
move 'javax.wsdl.importDocuments' to a constant

Reply:
We need to include below code segment as well. Otherwise, WSDLs with external dependencies will fail.

```
 			if (wsdlReader instanceof WSDLReaderImpl) {
 			    ((WSDLReaderImpl) wsdlReader).setIgnoreSchemaContent(true);
 			}
```
https://github.com/wso2/carbon-apimgt/blob/v6.2.76/components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIMWSDLReader.java#L255-L257